### PR TITLE
Enable test coverage for pocketcasts target

### PR DESCRIPTION
**Pull request by Deco354**
## Changes

1.  Enable xcode test coverage reports
2.  Enable xcode side bar test coverage

| 1 | 2 |
|:-------:|:-------:|
| image1 | image2 |

## Why

Apple's test coverage reports allow us to see at a glance what code has been tested and where we're lacking test coverage. The overall figure is also a valuable metric to track whether sufficient testing is being done for a given project, component or class which helps inform team policy decisions regarding testing or testability enablers such as modularization, or dependency injection.

The side bar test coverage can help developers when writing tests to see what parts of a class they're not testing and give us ideas for additional tests.

While it's a valuable metric/developer tool it shouldn't be used as a golden bullet. The absence of coverage is useful in itself but just because code is covered doesn't mean it's well tested as we need to test the code's behavior in a variety of circumstances and this involves writing much more than a single test per function. I'm also of the opinion that 100% coverage is not something any healthy project should aim for.

## To test

* Run all unit tests
* No new warnings or errors should appear
* Go to the report navigator (cmd 9) and click coverage. There should be a full test coverage report showing 1.5% total coverage.
* Check that 3rd party code has not been included within the coverage figures
* Check that all 1st party code has been included within the coverage figures

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Review by leandroalonso — CHANGES_REQUESTED** [review]
Hey @Deco354, thanks for submitting this PR.

Before we move on, could you please update the PR description with more details? What's the purpose of enabling test coverage?

Thanks!

File: PocketCastsTests/UnitTests.xctestplan [hunk 1]
```diff
@@ -13,7 +13,65 @@
     }
   ],
   "defaultOptions" : {
-    "codeCoverage" : false,
+    "codeCoverage" : {
```
Comment (leandroalonso) [inline PocketCastsTests/UnitTests.xctestplan]: What do you think about:

1. Keeping `podcasts`
2. Keeping `Pocket Casts Watch App`
3. Removing the rest
4. Adding `PocketCastsDataModel`
5. Adding `PocketCastsServer`
6. Adding `PocketCastsUtil`

The reasoning for that is:

1. I feel it's easier to follow code coverage for a few targets. If we have too many of them, it's hard to follow — because it's too much
2. `DataModel`,  `Server` and `Utils` are probably the targets in which tests are really important, given that there's no UI and it should be easier to test (well, a lot of it in practice it's not but that's another story)

Let me know what do you think.

**Review by leandroalonso — CHANGES_REQUESTED** [review]

File: PocketCastsTests/UnitTests.xctestplan (hunk 1 above)
Comment (Deco354) [reply PocketCastsTests/UnitTests.xctestplan]: > Adding PocketCastsDataModel
Adding PocketCastsServer
Adding PocketCastsUtil

Good catch I missed these 👍

> Removing the rest

It seem's there [an issue](https://developer.apple.com/forums/thread/133517) with getting a packages test coverage to display when selecting specific classes to get coverage for. At present to show a packages test coverage you need to get coverage for all targets within that scheme. (Useful to know for when we do this for Tumblr).

I think it's still worth doing as you still get the side bar coverage and could select a specific report if it's of interest to you. Although it does mean you need to disregard the total figure xcode provides as it includes Kingfisher and test packages.

Here's a screenshot.

image

File: podcasts.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 2]
```diff
@@ -42,17 +42,17 @@
...
+          "version": "0.9.1"
         }
       },
       {
         "package": "DifferenceKit",
         "repositoryURL": "https://github.com/ra1028/DifferenceKit",
         "state": {
           "branch": null,
-          "revision": "62745d7780deef4a023a792a1f8f763ec7bf9705",
-          "version": "1.2.0"
```
Comment (Deco354) [inline podcasts.xcworkspace/xcshareddata/swiftpm/Package.resolved]: @leandroalonso  I'm not really sure why this has happened, I've not used swiftpm much.
Is this normal or have I triggered an update some how 🤔

Comment (leandroalonso) [reply podcasts.xcworkspace/xcshareddata/swiftpm/Package.resolved]: Hmm, you probably triggered an update. Could you please revert this file?

**Comment by Deco354** [comment]
I was a little worried this change might impact build times. 
I ran a build time summary for clean-builds with and without coverage to check if there's a significant build time regression while test coverage is enabled.

Without test coverage a clean-build completed in 79.8 seconds
With test coverage a clean-build completed in 80.1 seconds

[test-cov-build-time-summary.txt](https://github.com/Automattic/pocket-casts-ios/files/13820574/test-cov-build-time-summary.txt)
[no-test-cov-build-time-summary.txt](https://github.com/Automattic/pocket-casts-ios/files/13820575/no-test-cov-build-time-summary.txt)

**Comment by Deco354** [comment]
@leandroalonso This is ready for review, apologies for the delay, I've been afk most of December and had a lot to do before leaving Tumblr.

CI is failing although it appears to be an issue with CI not connecting to its xcode image rather than the contents of this PR.

**Comment by pocketcasts** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by leandroalonso — APPROVED** [review]
